### DISCUSSION TITLE
CGPR1 motion sensor inconsistent battery report

### DISCUSSION THREAD

**Discussion by hugalafutro** [custom-components/ble_monitor#557] [Q&A]
Hi,
I've noticed that my CGPR1 motion sensor reports battery as such:
![Screenshot 2021-11-13 100424](https://user-images.githubusercontent.com/30209689/141614614-044e2d54-920f-46f5-8937-be87f12d61b4.png)
which sometimes triggered my automation to warn me battery in iot device is going low. I selected "use median" for the particular sensor in the integration options, but the graph still looks wonky.

Just wondering if others with this sensor see the same by any chance and could this be indicative of the actual battery being close to dying?

edit: larger history shows this repeats cyclically:
![Screenshot 2021-11-13 100903](https://user-images.githubusercontent.com/30209689/141614730-9279e6f9-8bf6-4f1d-a709-de99ed349e4c.png)

**Comment by Ernst79** [comment]
hmm, very strange. I don't have this sensor myself, and haven't seen this before. I'll move this to a new issue. If others are also facing this issue, let me know. 

I'm working on implementing the qingping advertisement format at the moment for the CGPR1. This sensor supports two types of advertisements, MiBeacon when connected to Mi Home, and Qingping format otherwise. Perhaps this Qingping format works better.
